### PR TITLE
Migrate github-release pipeline to 1ES governed template

### PR DESCRIPTION
**Pull request by shishirb-MSFT**

### CODE REVIEW

File: azure-pipelines/build/linux/du/templates/do-docker-jobs.yml [hunk 1]
```diff
@@ -17,6 +17,8 @@ jobs:
       targetOsArch: ${{parameters.targetOsArch}}
       imageVersion: ${{parameters.imageVersion}}
       config: debug
+  - publish: $(Build.ArtifactStagingDirectory)
+    artifact: 'doclient-drop-${{parameters.targetOsArch}}-debug'
```
Comment (Jerry-GeGarcia) [inline azure-pipelines/build/linux/du/templates/do-docker-jobs.yml:21]: >drop [](http://example.com/codeflow?start=24&length=4)

is this naming convention new?

File: azure-pipelines/build/linux/du/templates/doclient-lite-docker-steps.yml [hunk 2]
```diff
@@ -49,11 +49,5 @@ steps:
     SourceFolder: '/tmp/build-deliveryoptimization-agent-${{parameters.targetOsArch}}/linux-${{parameters.config}}'
     Contents: |
       client-lite/deliveryoptimization-agent.dbg
-    TargetFolder: '$(Build.ArtifactStagingDirectory)/${{parameters.targetOsArch}}-${{parameters.config}}'
+    TargetFolder: '$(Build.ArtifactStagingDirectory)/agent-${{parameters.targetOsArch}}-${{parameters.config}}'
   displayName: 'Populate debug symbols to staging dir'
-
-- task: PublishBuildArtifacts@1
```
Comment (Jerry-GeGarcia) [inline azure-pipelines/build/linux/du/templates/doclient-lite-docker-steps.yml:55]: >PublishBuildArtifacts [](http://example.com/codeflow?start=8&length=21)

this was removed across all three. I assume the "artifact" part in the do-docker-jobs.yml does the equivalent now?

File: azure-pipelines/publishing/github-release.yml [hunk 3]
```diff
@@ -6,140 +6,183 @@ trigger:
...
-          $(Build.ArtifactStagingDirectory)/hashes*.txt
-        tagSource: 'userSpecifiedTag'
-        tag: '$(Release.Version)'
-        title: '$(Release.Title)'
-        isPreRelease: true
-        isDraft: ${{parameters.PublishAsDraft}}
-        changeLogCompareToRelease: 'lastNonDraftReleaseByTag'
-        changeLogCompareToReleaseTag: '$(Release.PreviousVersion)'
-        changeLogType: 'commitBased'
+- name: SkipPublishing
```
Comment (Jerry-GeGarcia) [inline azure-pipelines/publishing/github-release.yml:16]: >SkipPublishing [](http://example.com/codeflow?start=8&length=14)

is this where we will integrate the PMC publishing flow?

**Comment by Jerry-GeGarcia** [comment]
@shiyi-peng is added to the review. #Closed

File: build/docker/ubuntu1804/amd64/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Dockerfile for building DO client components for Ubuntu 18.04 x64
+# First, install the docker extension for VSCode. Then you can right-click on this file
+# and choose Build Image. Give it a name and it will build the image.
+#
+# Open interactive terminal into the image in a container:
+# docker run -ti --rm --entrypoint=/bin/bash -v <project root dir>:/code -v <build root dir>:/build <image_name>
+# Example:
+# docker run -ti --rm --entrypoint=/bin/bash -v D:\do-client-lite:/code -v D:\temp\build_client_lite\linux-debug:/build ubuntu1804-x64
+
+FROM mcr.microsoft.com/mirror/docker/library/ubuntu:18.04@sha256:152dc042452c496007f07ca9127571cb9c29697f42acbfad72324b2bb2e43c98
```
Comment (Jerry-GeGarcia) [inline build/docker/ubuntu1804/amd64/Dockerfile:10]: >152dc042452c496007f07ca9127571cb9c29697f42acbfad72324b2bb2e43c98 [](http://example.com/codeflow?start=65&length=64)

hash of the container? worth adding a comment?

**Review by Jerry-GeGarcia — APPROVED** [review]
:shipit:

File: azure-pipelines/build/linux/du/templates/do-docker-jobs.yml (hunk 1 above)
Comment (shishirb-MSFT) [reply azure-pipelines/build/linux/du/templates/do-docker-jobs.yml:21]: Yes. The 1ES tasks also publish artifacts/drop, so I'm making ours distinguishable.

File: azure-pipelines/build/linux/du/templates/doclient-lite-docker-steps.yml (hunk 2 above)
Comment (shishirb-MSFT) [reply azure-pipelines/build/linux/du/templates/doclient-lite-docker-steps.yml:55]: Yes. And the same part in release-docker-build-steps.yml as well.

It was the migration tool that moved these lines from here the parent yaml, so I kept it that way.

File: azure-pipelines/build/linux/du/templates/do-docker-jobs.yml (hunk 1 above)
Comment (Jerry-GeGarcia) [reply azure-pipelines/build/linux/du/templates/do-docker-jobs.yml:21]: I see. What is different from the artifacts that the 1ES task produces and ours?

File: azure-pipelines/publishing/github-release.yml (hunk 3 above)
Comment (shishirb-MSFT) [reply azure-pipelines/publishing/github-release.yml:16]: We are currently attempting to create that as its own separate pipeline (release pipeline) but hitting some issues. We'll know more next week.

File: azure-pipelines/build/linux/du/templates/do-docker-jobs.yml (hunk 1 above)
Comment (shishirb-MSFT) [reply azure-pipelines/build/linux/du/templates/do-docker-jobs.yml:21]: Example artifacts: [https://dev.azure.com/deliveryoptimization/client/_build/results?buildId=4469&view=artifacts&pathAsName=false&type=publishedArtifacts](https://dev.azure.com/deliveryoptimization/client/_build/results?buildId=4469&view=artifacts&pathAsName=false&type=publishedArtifacts)

Comment (Jerry-GeGarcia) [reply azure-pipelines/build/linux/du/templates/do-docker-jobs.yml:21]: thank you, that helps. The doclient convention makes sense; wasn't too sure about the "drop", but I guess its consistent.

**Pull request merged**